common/views/components/Modal/Modal.tsx

```diff
@@ -271,7 +271,10 @@ const Modal: FunctionComponent<Props> = ({
                 <Icon icon={cross} />
               </CloseButton>
             )}
-            {children}
+            {/* The Modal element needs to be pre-rendered even if inactive for its CSSTransition effect
+            But there's a bit of rerending that's possible within its children that could affect the pages
+            even when the modal is not active, so hiding if not. */}
+            {isActive && children}
```

Comment:
Right. @georgiaewhitney / @jamieparkinson:

I started [this chat on Slack](https://wellcome.slack.com/archives/C3TQSF63C/p1702906257849789), so linking for some context.

Basically the [ColorPaletteFilter in the Modal box was re-rendering on `hasNoResult` state change](https://github.com/wellcomecollection/wellcomecollection.org/blob/main/content/webapp/components/ModalMoreFilters/index.tsx#L188). This caused issues even when the modal wasn't active, because as far as I can understand, even if it doesn't display, it is rendered in the background, and once `isActive` is `true`, it uses `CSSTransition` to display with a bit of style. 
So the workaround was to block the inner content of the modal from being rendered instead, if the modal is not active. [I originally did it only in the Filters modal](https://github.com/wellcomecollection/wellcomecollection.org/pull/10325/commits/c3fc9605ae0472d3bb65f5418ba19c21810a09db) and then thought maybe it could be applied across all modals (as it is here now).

That's my end of day brain though. I'd love your opinions on this. @jamieparkinson, I believe the re-render happened because of the [useEffects in PaletteColorPicker](https://github.com/wellcomecollection/wellcomecollection.org/blob/main/content/webapp/components/PaletteColorPicker/PaletteColorPicker.tsx#L162) and I wondered if you remembered the logic or maybe if it was worth tweaking that instead? Although it feels logical to me now to block all inner contents if not required to display.

LMK!

Reply:
OK. Sigh. I think this fix breaks things in mobile as the filters depend solely on the modal retaining the information. 
I think I'll go back to [my first iteration](https://github.com/wellcomecollection/wellcomecollection.org/pull/10325/commits/c3fc9605ae0472d3bb65f5418ba19c21810a09db) and only apply it where it breaks thing (desktop version).